### PR TITLE
v3: get rid of org.json and use Gson for performance

### PR DESCRIPTION
**Pull request by gfx**
https://github.com/cookpad/puree-android/issues/31

* [x] get rid of org.json
* [x] benchmark with v2 - about 2 times faster in average
  * I think it is good enough
* [x] documentation

## benchmark

Note it is not fair enough because `Puree.flush()` invokes asynchronous tasks. That is because the first try is so fast. 

v2
screenshot_2015-06-09-09-01-58

v3
screenshot_2015-06-09-09-05-38

### CODE REVIEW

File: demo/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
     <!-- Base application theme. -->
     <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
-        <!-- Customize your theme here. -->
+        <item name="android:textColor">#ffffffff</item>
```
Comment (rejasupotaro) [inline demo/src/main/res/values/styles.xml]: [nits] should use `@android:color/white`.

File: demo/src/main/java/com/example/puree/BenchmarkActivity.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+public class BenchmarkActivity extends AppCompatActivity {
+
+    ArrayList<SendLogs> tasks = new ArrayList<>();
+
+
```
Comment (rejasupotaro) [inline demo/src/main/java/com/example/puree/BenchmarkActivity.java]: [nits] delete empty line.

**Comment by rejasupotaro** [comment]
Good improvement :+1:

**Comment by gfx** [comment]
Thanks!

**Comment by gfx** [comment]
@rejasupotaro

One more thing: changed `class JsonConvertible` into `interface PureeLog` to make things clear.

**Comment by rejasupotaro** [comment]
@gfx OK, I noted.

**Comment by gfx** [comment]
It's time to merge!

**Pull request merged**